mm/kfence/core.c

```diff
@@ -560,6 +560,34 @@ bool kfence_discard_slab(struct kmem_cache *s, struct page *page)
 	return true;
 }
 
+bool kfence_shutdown_cache(struct kmem_cache *s)
+{
+	unsigned long flags;
+	int i;
+	struct kfence_alloc_metadata *meta;
+	bool ret = false;
+
+	spin_lock_irqsave(&kfence_alloc_lock, flags);
+
+	for (i = 0; i < KFENCE_NUM_OBJ; i++) {
+		meta = &kfence_metadata[i];
+		if ((meta->cache == s) && (meta->state == KFENCE_OBJECT_ALLOCATED))
+			goto leave;
+	}
+
+	for (i = 0; i < KFENCE_NUM_OBJ; i++) {
+		meta = &kfence_metadata[i];
+		if ((meta->cache == s) && (meta->state == KFENCE_OBJECT_FREED))
+			meta->cache = NULL;
+	}
+	ret = true;
+
+leave:
+	spin_unlock_irqrestore(&kfence_alloc_lock, flags);
+	return ret;
+}
+EXPORT_SYMBOL(kfence_shutdown_cache);
```

Comment:
I don't think modules should use this, right? I'd expect this to only be used from allocator internals.

Reply:
Ah, correct, it is called from a static function.

Reply:
Fixed.